geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyGatewayReceiverCommand.java

```diff
@@ -68,7 +68,9 @@ public Result destroyGatewayReceiver(
     // Only update the cluster config if the command is not executed on specific members.
     InternalClusterConfigurationService service =
         (InternalClusterConfigurationService) getConfigurationService();
-    if (onMember == null && result.getStatus().equals(Result.Status.OK) && service != null) {
+    if (onMember != null || service == null) {
+      persisted = false;
+    } else if (result.getStatus().equals(Result.Status.OK)/* && service != null */) {
```

Comment:
This check should remain in place.

Reply:
Good catch. Left over from some debug iterations. I'll fix this along with the conflicts resulting from renaming `ClusterConfigurationService`.